desc/optimize/optimizer.py

```diff
@@ -163,40 +162,46 @@ def optimize(  # noqa: C901 - FIXME: simplify this
         # TODO: document options
         timer = Timer()
         options = {} if options is None else options
-        wrapper, method = _parse_method(self.method)
+        _, method = _parse_method(self.method)
 
-        linear_constraints, nonlinear_constraint = _parse_constraints(constraints)
-        objective, nonlinear_constraint = _maybe_wrap_nonlinear_constraints(
-            eq, objective, nonlinear_constraint, self.method, options
+        # parse and combine constraints into linear & nonlinear objective functions
+        linear_constraints, nonlinear_constraints = _parse_constraints(constraints)
+        objective, nonlinear_constraints = _maybe_wrap_nonlinear_constraints(
+            eq, objective, nonlinear_constraints, self.method, options
         )
+        if not isinstance(objective, ProximalProjection) and eq is not None:
+            linear_constraints = maybe_add_self_consistency(eq, linear_constraints)
+        linear_constraint = _combine_constraints(linear_constraints)
+        nonlinear_constraint = _combine_constraints(nonlinear_constraints)
 
         # make sure everything is built
         if objective is not None and not objective.built:
             objective.build(verbose=verbose)
+        if linear_constraint is not None and not linear_constraint.built:
+            linear_constraint.build(verbose=verbose)
         if nonlinear_constraint is not None and not nonlinear_constraint.built:
             nonlinear_constraint.build(verbose=verbose)
 
+        # combine arguments from all three objective functions
         if nonlinear_constraint is not None:
-            objective, nonlinear_constraint = combine_args(
-                objective, nonlinear_constraint
+            objective, linear_constraint, nonlinear_constraint = combine_args(
```

Comment:
dont we still need to combine args for the objective and linear constraints, even if there aren't any nonlinear constraints?

Reply:
Yeah I guess none with more than 1 thing and no nonlinear constraints?